kratos/expression/expression.h

```diff
@@ -187,6 +187,19 @@ class KRATOS_API(KRATOS_CORE) Expression {
      */
     IndexType GetItemComponentCount() const;
 
+    /**
+     * @brief Get the Flat Expression
+     *
+     * Returns an expression containing a flattened expression. This will always
+     * be a LiteralFlatExpression. This keeps the item shape as it is.
+     *
+     * The flattened expression does not include any other expression combinations
+     * except the LiteralFlatExpression.
+     *
+     * @return Expression::ConstPointer     Flattened expression
+     */
+    ConstPointer Flatten() const;
```

Comment:
The `Flatten` name is very confusing because it is already used for converting nested vectors/matrices into contiguous arrays. I'd rather call it `Collapse`, but I'm open to other suggestions too.

Reply:
On a more general note, I don't think these kinds of functions should be members of `Expression`. They behave more like operators (produce a new `Expression` and don't mutate the host object), so they should be free-floating functions (or moved to some utility class).

Reply:
Agree... Will move now :)